services/src/main/java/io/druid/cli/QueryJettyServerInitializer.java

```diff
@@ -63,7 +69,18 @@ public void initialize(Server server, Injector injector)
     final ServletContextHandler root = new ServletContextHandler(ServletContextHandler.SESSIONS);
     root.addServlet(new ServletHolder(new DefaultServlet()), "/*");
 
-    final AuthConfig authConfig = injector.getInstance(AuthConfig.class);
+    // Add LimitRequestsFilter as first in the chain if enabled.
+    if (serverConfig.isEnableRequestLimit()) {
+      Preconditions.checkArgument(
+          serverConfig.getNumThreads() > 1,
+          "numThreads must be > 1 to enable Request Limit Filter."
+      );
+      log.info("Enabling Request Limit Filter with limit [%d].", serverConfig.getNumThreads()-1);
+      root.addFilter(new FilterHolder(new LimitRequestsFilter(serverConfig.getNumThreads()-1)),
+                     "/*", null
```

Comment:
Some of the "management" endpoints (like lookups) have a thread limit already. Having them both collide could be confusing.

Reply:
not sure about that but it should be simple to understand. this is a server level hard limit.... finer end point specific limits are there for resource appropriation really.